text/0939-import-glob.md

```diff
@@ -0,0 +1,196 @@
+
+ - However, due to the way template-only components work, you should think of this happening *after* the automatically-created Javascript representation of a template-only component is created. That is, if you want to import a directory full of components, even if some of them are template-only components represented by `.hbs` files, you should still `import.meta.glob('./components/*.js')` as that will match the automatically-created components.
+
+    > An import with an explicit `.hbs` extension has a specific historical meaning that is *not* a component, it's a bare template. You almost never want to do that manually. It's an implementation detail of template co-location and a historical compatibility feature. So if you tried to do `import.meta.glob('./components/*.{js,hbs}')` you would get back a mix of components and things-that-are-not-really-components.
+    >
+    > Ultimately this is a concern that goes away once you adopt `.gjs` and that would be our recommendation going forward.
+
+`*` matches everything except:
+ - path separators 
+ - names starting with "." (hidden files)
+
+`**` matches zero or more directories
+
+`?` matches any single character except path separators
+
+`[abc]`: a sequence of characters inside `[]` match any character in that sequence
+
+`{.js,.gjs}`: bash-style brace expansion
+
+### Lazy Loading
+
+`import.meta.glob()` is designed to work nicely in systems that lazily load code. However, it does not promise to *introduce* laziness where laziness does not already exist. 
+
+When building with the classic build pipeline, all your own app code is always included in the bundle regardless of whether anyone imports it or not, and that remains true regardless of whether you use `import.meta.glob()` to access some modules. 
+
+When building with Embroider, you can achieve lazy loading by using `import.meta.glob()` in combination with other features like `staticAppPaths` or `staticComponents`.
+
+The return value from `import.meta.glob()` is the same either way -- you always get functions that return `Promise<Module>`. 
+
+### Replacing cross-package usages
+
+Historically, people have used `requirejs.entries` to have complete global access to everywhere from everywhere. `import.meta.glob` is deliberately more restrictive. For example, an addon cannot use `import.meta.glob` to load code out of the consuming application. Instead, addon authors will need to ask apps to pass them what they need.
+
+For example, a future version of `ember-cli-mirage` might tell app authors to put this code into their app and/or tests as a way to dynamically gain access to all the Mirage-specific models, adapters, serializers, etc that the user has written:
+
+```js
+import { setup } from 'ember-cli-mirage';
+
+setup(import.meta.glob('./mirage/**/*.js'))
```

Comment:
this would immensely help the migration path to `ember-mirage`

cc @bgantzler / @cah-brian-gantzler

Reply:
Not only the migration path, but if its an approved way to discover files, could put the auto import of all the mirage files back in instead of you having to import them individually and create the hash